internal/backends/collection.go

```diff
@@ -79,8 +79,10 @@ type SortField struct {
 
 // QueryParams represents the parameters of Collection.Query method.
 type QueryParams struct {
-	Filter        *types.Document
-	Sort          *SortField
+	Filter       *types.Document
+	Sort         *SortField
+	SortPushdown bool
```

Comment:
There is no documentation update or contract change. For someone implementing a new backend, it is not possible to understand what those fields mean, what different combinations of them mean, how they should be used, etc.

Reply:
Should we document all fields, or only those "less obvious"?

Reply:
Not sure how well I've understood. PTAL at the latest commits